src/containerapp/azext_containerapp/_validators.py

```diff
@@ -50,9 +50,15 @@ def validate_create(registry_identity, registry_pass, registry_user, registry_se
         raise InvalidArgumentValueError("--registry-identity: expected an ACR registry (*.azurecr.io) for --registry-server")
 
 
-def validate_runtime(runtime, enable_java_metrics):
-    if runtime and runtime.lower() == RUNTIME_GENERIC and enable_java_metrics is not None:
-        raise ValidationError("Not support enable Java metrics with --enable-java-metrics for generic runtime with --runtime generic")
+def validate_runtime(runtime, enable_java_metrics, enable_java_agent):
+    def is_java_enhancement_enabled():
+        return enable_java_agent is not None or enable_java_metrics is not None
+
+    if runtime is None:
+        return
+    if runtime.lower() == RUNTIME_GENERIC and is_java_enhancement_enabled():
+        raise ValidationError("Not support enable Java metrics or Java agent enhancement "

```

Comment:
Recommended error message: `Usage error: --runtime java is required when using --enable-java-metrics or --enable-java-agent` which might be more user friendly

Reply:
Help me pls